### PR TITLE
adds v2 of final report

### PR DESCRIPTION
**Pull request by Zorrow2008**
This PR adds a new version of the final report that:
- goes in depth about each feature
- gives more info about cpu state
- gives more info about message passing

### CODE REVIEW

File: public/assets/reports/report.tex [hunk 1]
```diff
@@ -131,7 +132,7 @@ \section{Initial Design}
 
 To start, we stripped the riscv-angel project to only show the Terminal.
 We got rid of all the elements and libraries that were not going to be used in our project.
-Before we started on our implementation, we prototyped the UI in Figma, shown in Figure \ref{fig:proto1}.
+Before we started on our implementation, we prototyped the UI in Figma, shown in Figure 1.
```
Comment (swkeever) [inline public/assets/reports/report.tex:135]: how come you got rid of the the reference?

File: public/assets/reports/report.tex [hunk 2]
```diff
@@ -142,7 +143,9 @@ \section{Initial Design}
 
 Figma allowed us to come up with a design before we implement anything in code.
 We aimed for a dashboard-like interface, showing the user
-information about the CPU and OS in real-time.
+information about the CPU and OS in real-time. The dashboard interface would
+allow the user to have all the information at his or her fingertips without
```
Comment (swkeever) [inline public/assets/reports/report.tex]: could we say just "his" or just "her" or even "their" lol sometimes "his or her" seems a little verbose. not super important, more of a nit pick

File: public/assets/reports/report.tex [hunk 3]
```diff
@@ -185,17 +188,20 @@ \subsection*{Setting up a development environment}
 \subsection*{Choosing technologies}
 
 We want our interface to be interactive, so we chose React.js as our library of choice,
-which offers a means of rapidly developing interactive client-side applications.
+which offers a means of rapidly developing interactive client-side applications. React's
+rendering mechanism allows for changes in state to immediately update on the browser,
+which provides a user-friendly webpage.
```
Comment (swkeever) [inline public/assets/reports/report.tex]: i'm not sure this implication is necessarily true. what about something along the lines of

which allowed us to quickly create interactive UI components

or something?

File: public/assets/reports/report.tex [hunk 4]
```diff
@@ -262,21 +268,41 @@ \subsection*{A Modular Design}
 worrying about conflicting with another person's work.
 
 Finally, having the components act as modular entities allows us to change the layout of the UI very easily should
-we choose too. The components are simply independent building blocks that can be moved around as we please. This
+we choose too. The components are simply independent building blocks that can be moved around as we please.
```
Comment (swkeever) [inline public/assets/reports/report.tex]: remove "simply"

File: public/assets/reports/report.tex [hunk 5]
```diff
@@ -262,21 +268,41 @@ \subsection*{A Modular Design}
...
+Component indepedence is very useful for debugging purposes as well, as if one specific component isn't behaving
```
Comment (swkeever) [inline public/assets/reports/report.tex]: spelling of independence

File: public/assets/reports/report.tex [hunk 6]
```diff
@@ -262,21 +268,41 @@ \subsection*{A Modular Design}
...
+as intended, then a developer can isolate the problem to that component alone. This
 flexibility will become very important when it comes time to finalizing our app.
 
 \subsection*{Finishing our Prototype}
 
+There are two key files in the vanilla Javascript files that direct boot-up. Boot.js is run on start-up
```
Comment (swkeever) [inline public/assets/reports/report.tex]: in latex, you can write filenames in monospace with `\texttt{boot.js}`. this is a little cleaner way to do it imo

File: public/assets/reports/report.tex [hunk 7]
```diff
@@ -287,7 +313,25 @@ \subsection*{Finishing our Prototype}
 \end{figure}
 
 \noindent
-the ratio of CPU instructions executed,
+
+As Figure 3 shows, the register panel shows the register name, ABI name, and the state of
+each register. The register value will be displayed in 64bit Unsigned Hexadecimal.
```
Comment (swkeever) [inline public/assets/reports/report.tex]: unsigned hexadecimal

File: public/assets/reports/report.tex [hunk 8]
```diff
@@ -287,7 +313,25 @@ \subsection*{Finishing our Prototype}
...
+We partition the registers into three categories: All, callee-saved, and caller-saved.
```
Comment (swkeever) [inline public/assets/reports/report.tex]: should we make a latex unordered list to list the categories?

File: public/assets/reports/report.tex [hunk 9]
```diff
@@ -287,7 +313,25 @@ \subsection*{Finishing our Prototype}
...
+The react hook use-cpu will query for the latest cpu state from the Window object, and set the local
```
Comment (swkeever) [inline public/assets/reports/report.tex]: React

File: public/assets/reports/report.tex (hunk 1 above)
Comment (Zorrow2008) [reply public/assets/reports/report.tex:135]: it shows up as "??" on my end, so just wrote 1 for simplicity

File: public/assets/reports/report.tex (hunk 6 above)
Comment (Zorrow2008) [reply public/assets/reports/report.tex]: changed

**Pull request merged**